flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -932,7 +932,7 @@ private void resetAndStartScheduler() throws Exception {
 			);
 		}
 
-		schedulerAssignedFuture.thenRun(this::startScheduling);
+		FutureUtils.assertNoException(schedulerAssignedFuture.thenRun(this::startScheduling));
```

Comment:
Out of curiosity: What speaks against additionally changing `SchedulerBase.startAllOperatorCoordinators()` as follows:

```
	private void startAllOperatorCoordinators() {
		final Collection<OperatorCoordinatorHolder> coordinators = getAllCoordinators();
		try {
			for (OperatorCoordinatorHolder coordinator : coordinators) {
				coordinator.start();
			}
		}
		catch (Throwable t) {
			ExceptionUtils.rethrowIfFatalErrorOrOOM(t);
			coordinators.forEach(IOUtils::closeQuietly);
			failJob(new FlinkRuntimeException("Failed to start the operator coordinators", t)); // <-- change here
		}
	}
```

Reply:
The intention of this PR is to fix the problem of swallowed exceptions and not the job lifecycle management based on operator coordinator failures. I don't know whether restarting a job makes sense if the operator coordinator has failed to start. To me it looks as if the operator coordinator integration hasn't been completed yet. I would consider this as a follow up task.